File: event-service/models/event.go
package models

type Event struct {
	ID             int64  `db:"id" json:"-"`
	PublicID       string `db:"public_id" json:"id"`
	OrganizationID string `db:"organization_id" json:"organization_id"`
	Name           string `db:"name" json:"name"`
	Description    string `db:"description" json:"description"`
	StartDate      string `db:"start_date" json:"start_date"`
	EndDate        string `db:"end_date" json:"end_date"`
	VenueName      string `db:"venue_name" json:"venue_name"`
	VenueAddress   string `db:"venue_address" json:"venue_address"`
	VenueCity      string `db:"venue_city" json:"venue_city"`
	VenueCountry   string `db:"venue_country" json:"venue_country"`
	VenueCapacity  int    `db:"venue_capacity" json:"venue_capacity"`
	CanvasConfig   string `db:"canvas_config" json:"canvas_config"`
	Status         string `db:"status" json:"status"`
	EventType      string `db:"event_type" json:"event_type"`
	Category       string `db:"category" json:"category"`
	SaleStartDate  string `db:"sale_start_date" json:"sale_start_date"`
	SaleEndDate    string `db:"sale_end_date" json:"sale_end_date"`
	MinAge         int    `db:"min_age" json:"min_age"`
	IsFeatured     bool   `db:"is_featured" json:"is_featured"`
	Images         string `db:"images" json:"images"`
	Tags           string `db:"tags" json:"tags"`
	Metadata       string `db:"metadata" json:"metadata"`
	CreatedAt      string `db:"created_at" json:"created_at"`
	UpdatedAt      string `db:"updated_at" json:"updated_at"`
} 

File: frontend/src/lib/validators/event.ts
import { z } from "zod";

export const createEventSchema = z.object({
  name: z.string().min(2, "Event name must be at least 2 characters"),
  description: z.string().optional(),
  start_date: z.string().min(1, "Please select a start date"),
  end_date: z.string().min(1, "Please select an end date"),
  venue_name: z.string().min(2, "Venue name must be at least 2 characters"),
  venue_address: z.string().optional(),
  venue_city: z.string().optional(),
  venue_country: z.string().optional(),
  venue_capacity: z.string().optional(),
  event_type: z.string().optional(),
  category: z.string().optional(),
  sale_start_date: z.string().optional(),
  sale_end_date: z.string().optional(),
  min_age: z.string().optional(),
  is_featured: z.boolean().optional(),
});

export type CreateEventInput = z.infer<typeof createEventSchema>;

export const ticketTypeSchema = z.object({
  name: z.string().min(1, "Ticket type name is required"),
  description: z.string().optional(),
  price: z.number().min(0, "Price must be 0 or more"),
  currency: z.string().optional(),
  quantity: z.number().int().min(1, "At least 1 ticket required"),
  max_per_purchase: z.number().int().min(1).optional(),
  min_per_purchase: z.number().int().min(1).optional(),
});

export type TicketTypeInput = z.infer<typeof ticketTypeSchema>;
